src/composables/use-search.ts

```diff
@@ -0,0 +1,89 @@
+import { computed, ref, useRouter, watch } from "@nuxtjs/composition-api"
+
+import { useSearchStore } from "~/stores/search"
+import { useMediaStore } from "~/stores/media"
+import { useI18nResultsCount } from "~/composables/use-i18n-utilities"
+import { useMatchSearchRoutes } from "~/composables/use-match-routes"
+import { ALL_MEDIA } from "~/constants/media"
+
+export const useSearch = () => {
```

Comment:
The amount of component-level code that this composable simplifies is _excellent_ :rocket: Nice work!